packages/blade/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -113,13 +122,36 @@ const ProgressBar = ({
   ...styledProps
 }: ProgressBarProps): ReactElement => {
   const { theme } = useTheme();
-  const id = useId(variant);
+  const progressType = !type && (variant === 'meter' || variant === 'progress') ? variant : type;
+  const progressVariant = variant === 'meter' || variant === 'progress' ? 'linear' : variant;
+  const id = useId(`${progressType}-${progressVariant}`);
 
   if (__DEV__) {
-    if (variant === 'meter' && isIndeterminate) {
+    if (progressType === 'meter' && isIndeterminate) {
+      throwBladeError({
+        moduleName: 'ProgressBar',
+        message: `Cannot set 'isIndeterminate' when 'type' is 'meter'.`,
+      });
+    }
+
+    if (progressVariant === 'circular' && isIndeterminate) {
+      throwBladeError({
+        moduleName: 'ProgressBar',
+        message: `Cannot set 'isIndeterminate' when 'variant' is 'circular'.`,
```

Comment:
So we aren't sufficing the spinner usecase with Circular progressbar is it?

Reply:
Yes. The Spinner will be a separate component.